include/deal2lkit/parsed_zero_average_constraints.h

```diff
@@ -0,0 +1,169 @@
+//
+//    Copyright (C) 2016 by the deal2lkit authors
+//
+//    This file is part of the deal2lkit library.
+//
+//    The deal2lkit library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE at
+//    the top level of the deal2lkit distribution.
+//
+//-----------------------------------------------------------
+
+#ifndef _d2k_parsed_zero_average_constraints_h
+#define _d2k_parsed_zero_average_constraints_h
+
+#include <deal2lkit/config.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/parsed_function.h>
+#include <deal2lkit/parameter_acceptor.h>
+#include <deal.II/fe/component_mask.h>
+#include <algorithm>
+#include <map>
+#include <deal2lkit/utilities.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/lac/constraint_matrix.h>
+
+
+using namespace dealii;
+
+
+D2K_NAMESPACE_OPEN
+
+/**
+ * ParsedZeroAverageConstraints class. It allows you to set a zero
+ * average constraint to a ConstraintMatrix. The variable to be
+ * constrained (e.g. the pressure for Stokes problem) can be specified
+ * in the parameter file either by its component name (e.g. "p") or by
+ * the component number.
+ *
+ * The zero mean value can be set on the whole domain, or on the
+ * boundary.
+ *
+ * A typical usage of this class is as follows:
+ *
+ * @code
+ * ConstraintMatrix cm;
+ * ParsedZeroAverageConstraints<dim> pnac("Parsed Zero Average Constraints",
+ *              dim+1,
+ *            (dim==2?"u,u,p":"u,u,u,p"),
+ *            "p");
+ *
+ *
+ *
+ * ParameterAcceptor::initialize();
+ * pnac.apply_zero_average_constraints(dof_handler,cm);
+ * @endcode
+ */
+
+
+
+template <int dim, int spacedim=dim>
+class ParsedZeroAverageConstraints : public ParameterAcceptor
+{
+public:
+  /**
+   * Constructor.
+   *
+   * It takes:
+   * - the name for the section of the Parameter Handler to use
+   *
+   * - the number of components that this function has
+   *
+   * - the default names of known components that can be used instead
+   *   of component numbers
+   *
+   * - a list of components, separated by a "," (i.e. 0,p,5), that are
+   *   to be constrained on the whole domain
+   *
+   * - a list of components, separated by a "," (i.e. 0,p,5), that are
+   *   to be constrained on the boundary
+   */
+  ParsedZeroAverageConstraints  (const std::string &name,
+                                 const unsigned int &n_components=1,
+                                 const std::string &component_names = "",
+                                 const std::string &default_components = "",
+                                 const std::string &default_boundary_components ="");
+
+
+  /**
+   * This function must be called in order to apply the constraint on
+   * the domain.
```

Comment:
What happens if I call this function and no domain components where specified? I'd expect it to simply leave untouched the constraint matrix. If this is so, why do we need two different functions? Can't we just do everything on one function?